### PR TITLE
Add nixos service and scripts

### PR DESCRIPTION
**Pull request by rvl**
- [x] Adds a NixOS service module for cardano-wallet.
- [x] Adds a NixOS test for the service.
- [x] Documentation on how to use the NixOS service.

### Comments

This PR is @jbgi's work and was split off from #2997.

### Issue Number

ADP-1357

### CODE REVIEW

File: nix/scripts.nix [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{ evalService
```
Comment (rvl) [inline nix/scripts.nix:11]: I'm not so sure yet about these scripts.
Our users would like to have convenience scripts for connecting to mainnet/testtnet, but possibly in a way where nix is optional.
Can we just do the NixOS service in this PR, and review these convenience scripts in a new PR?

**Review by rvl — COMMENTED** [review]
Looks great @jbgi.
I can't :heavy_check_mark: because I opened this PR.

File: nix/scripts.nix (hunk 1 above)
Comment (jbgi) [reply nix/scripts.nix:11]: I think those scripts can/are also to be used as entry-point for bitte clusters containers, (cc @blaggacao for confirmation) so would be needed as part of the flake.

Comment (disassembler) [reply nix/scripts.nix:11]: Yes, these scripts are required to run the wallet in nomad clusters, such as ERC20 converter.

Comment (rvl) [reply nix/scripts.nix:11]: OK, thanks @jbgi, @disassembler.

**Pull request merged**